### ISSUE TITLE
CurrencyUnitDeserializer deserializes invalid currencies

### ISSUE THREAD

**Issue by ptahchiev** [zalando/jackson-datatype-money#34]
Hello,

at the moment the `CurrencyUnitDeserializer` will deserialize anything you pass as a currency. For example strings like `1123`, `eur`, or `xyz` are all invalid but the `CurrencyUnitDeserializer` will create a currency unit out of it. I think the correct behaviour is to only deserialize valid strings, such as `EUR` or `USD`. This could be done with the following code:
````
    @Override
    public CurrencyUnit deserialize(final JsonParser parser, final DeserializationContext context) throws IOException {
        return Monetary.getCurrency(parser.getValueAsString());
    }

````

**Comment by whiskeysierra** [comment]
@ptahchiev Thanks for the report. I just released 0.8.0, should be available in central soon.

**Comment by whiskeysierra** [comment]
@ptahchiev I released 0.9.0 to central. Thanks for the contribution!
